python/GafferSceneUI/SceneHistoryUI.py

```diff
@@ -189,12 +197,47 @@ def __viewerKeyPress( viewer, event ) :
 def __hierarchyViewKeyPress( hierarchyView, event ) :
 
 	if event == __editSourceKeyPress :
-		selectedPath = __hierarchyViewSelectedPath( hierarchyView )
+		selectedPath = __contextSelectedPath( hierarchyView.getContext() )
 		if selectedPath is not None :
 			__editSourceNode( hierarchyView.getContext(), hierarchyView.scene(), selectedPath )
 		return True
 	elif event == __editTweaksKeyPress :
-		selectedPath = __hierarchyViewSelectedPath( hierarchyView )
+		selectedPath = __contextSelectedPath( hierarchyView.getContext() )
 		if selectedPath is not None :
 			__editTweaksNode( hierarchyView.getContext(), hierarchyView.scene(), selectedPath )
 		return True
+
+def __nodeEditorKeyPress( nodeEditor, event ) :
+
+	layout = nodeEditor.ancestor( GafferUI.CompoundEditor )
+	if layout is None :
+		return False
+
+	## \todo In Gaffer 0.61, we should get the scene directly from the focus node.
```

Comment:
I had to stare at this for a bit to understand what was even going on, but I eventually got it ... it's not about the node currently in the Node Editor, it's about pinning the node editor to what's currently in the HierarchyView or Viewer ... in the common case, hopefully the HierarchyView and Viewer are looking at the same thing, in which case this will work fine ( and using the Focus node in 61 will have the same effect ).  There are some weird corner cases though ... if someone sets the Focus node to a Catalog, and pins a Scene to the Viewer and Hierarchy, I'm not sure what the right answer is.  But probably focus node wins as being the most consistent ( even if not the most complete approach ).  Anyway, seems fine for now.  LGTM.